Sources/ClientRuntime/Identity/IdentityAPI/Identity.swift

```diff
@@ -0,0 +1,21 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// Base protocol for all identity types
+public protocol Identity {
+    var expiration: Date? { get }
+}
+
+// Identity v. IdentityT v. IdentityKind
+// - Identity is the protocol that all identity types must conform to.
+// - IdentityT is the associated type / generic type name used by protocols like IdentityResolver and Signer.
+// - IdentityKind is the enum that's used by IdentityResolverConfiguration to return correct kind of identity resolver
+//   for the given auth scheme. E.g., SigV4AuthScheme has idKind field as .aws. And identityResolver method in SigV4AuthScheme
+//   returns an identity resolver that returns identity of type .aws. 
+public enum IdentityKind: CaseIterable {
+    case aws
+}
```

Comment:
Clarifying question: Is an expiration all that would be needed to make something an Identity?

Reply:
It's just that expiration is the common property in all credentials. Each concrete implementation of this protocol will need to have variable(s) that store the identity data that they use. For example, AWS credentials struct that implements this protocol has variables for access key and secret key. Bearer token struct, when it's implemented, will implement this protocol and add a token variable that stores the token string.